int16 input1
int16 input2
